src/dove/core/components.py

```diff
@@ -189,6 +203,12 @@ class Source(Component):
     """
 
     def __init__(self, name: str, produces: Resource, **kwargs: Any) -> None:
+        for bad_kwarg in ("consumes", "capacity_resource"):
```

Comment:
might be better to provide a better error message. Source's can't consume and can't have a capacity_resource different from what it produces.